### PR TITLE
Media Capture and Streams plus MediaRecorder - sidebar fixes 

### PR DESCRIPTION
**Pull request by hamishwillee**
This fixes up links for the "Media Capture and Streams API" top level page by moving the page to have the same URL as the groupdata for the API. It also adds `Media Recorder API` entry to groupdata (note the top level overview page does not yet exist).

Last of all, this adds `{{APIRef}}` back to [MediaDevices.getUserMedia()](https://developer.mozilla.org/en-US/docs/Web/API/MediaDevices/getUserMedia) as the assertion was that this should be used rather than `{{DefaultAPISidebar}}` for API items in #18760

NOTE, this last point is problematic because with the `{{APIRef}}` no sidebar is displayed. Further, the `Media Recorder API` entry for BlobEvent which DID have a sidebar now does not. 
My assertion is that if you have groupdata for an api then the API ref does not work.

@teoli2003 My GUESS is that APIRef is for interfaces that do not have an overview page. Or it's just broken. What do you think we should do?

### CODE REVIEW

**Comment by hamishwillee** [comment]
Source of error is Windows rendering: https://github.com/mdn/yari/issues/6802

File: files/en-us/web/api/media_capture_and_streams_api/index.md [hunk 1]
```diff
@@ -11,16 +10,17 @@ tags:
   - Overview
   - Video
 browser-compat: api.MediaStream
+page-type: web-api-overview
 ---
-{{DefaultAPISidebar("Media Capture and Streams")}}
+{{APIRef("Media Capture and Streams")}}
```
Comment (hamishwillee) [inline files/en-us/web/api/media_capture_and_streams_api/index.md:15]: Except for this one!
```suggestion
{{DefaultAPISidebar("Media Capture and Streams")}}
```

File: files/en-us/web/api/media_capture_and_streams_api/taking_still_photos/index.md [hunk 2]
```diff
@@ -13,7 +13,6 @@ tags:
   - MediaDevices
   - getusermedia
   - webcam
```
Comment (hamishwillee) [inline files/en-us/web/api/media_capture_and_streams_api/taking_still_photos/index.md:15]: Add this back
```suggestion
  - webcam
page-type: guide
```

**Comment by hamishwillee** [comment]
@teoli2003 This is ready for review. It reverts the interface docs back to using APIRef. I did some messing with layout too, but no real content chagnes.

**Review by teoli2003 — APPROVED** [review]
Thank you and sorry for the mess…

**Comment by teoli2003** [comment]
I think this needs to be merged (I tried but failed miserabily)

**Comment by hamishwillee** [comment]
@teoli2003 - All fixed with a rebase. Thanks very much for catching this - I expect to inject some chaos into the docs, but not this much :-0

**Pull request merged**